AssetRipper.GUI/ConsoleApp.cs

```diff
@@ -1,126 +1,179 @@
 			{
-				if (MultiFileStream.Exists(arg))
-				{
+				if (type == null)
 					continue;
-				}
 
-				if (Directory.Exists(arg))
-				{
+				string optionName = ConsoleOptions.OptionNameFromType(type);
+				options.TryGetValue(optionName, out Option? option);
+
+				if (option == null)
 					continue;
-				}
 
-				System.Console.WriteLine(MultiFileStream.IsMultiFile(arg)
-					? $"File '{arg}' doesn't have all parts for combining"
-					: $"Neither file nor directory with path '{arg}' exists");
+				object? settingValue = context.ParseResult.GetValueForOption(option);
 
-				return false;
-			}
+				if (settingValue == null)
+					continue;
 
-			try
-			{
-				options.LogFile ??= new FileInfo(ExecutingDirectory.Combine(DefaultLogFileName));
-				options.OutputDirectory ??= new DirectoryInfo(ExecutingDirectory.Combine("Ripped"));
-			}
-			catch (Exception ex)
-			{
-				System.Console.WriteLine($"Failed to initialize the output and log paths.");
-				System.Console.WriteLine(ex.ToString());
-				return false;
+				string? enumKey = settingValue.ToString();
+
+				if (string.IsNullOrEmpty(enumKey))
+					continue;
+
+				ripper.Settings.SetSetting(type, Enum.Parse(type, enumKey));
 			}
 
-			return true;
+			DoRipping(ripper, inputs, output.FullName, verbose, logFile.FullName);
 		}
 
-		private static void Run(Options options)
+		private static void HandleException(Exception ex, InvocationContext context)
 		{
-			Logger.AllowVerbose = options.Verbose;
-			Logger.Add(new ConsoleLogger(false));
-			Logger.Add(new FileLogger(options.LogFile.FullName));
-			Logger.LogSystemInformation("AssetRipper Console Version");
-#if !DEBUG
-			try
-#endif
-			{
-				Ripper ripper = new();
-				ripper.Settings.LogConfigurationValues();
-				ripper.Load(options.FilesToExport);
-				PrepareExportDirectory(options.OutputDirectory.FullName);
-				ripper.ExportProject(options.OutputDirectory.FullName);
-			}
-#if !DEBUG
-			catch (Exception ex)
-			{
-				Logger.Log(LogType.Error, LogCategory.General, ex.ToString());
-			}
+			HandleException(ex);
+		}
+
+		private static void HandleException(Exception ex)
+		{
+			Environment.ExitCode = (int)ExitCode.ExtractingError;
+
+#if DEBUG
+			Console.WriteLine("===================================================");
+			Console.WriteLine("      Error during ripping, extraction aborted     ");
+			Console.WriteLine("===================================================");
+			Console.WriteLine(ex.ToString());
+#else
+			Console.WriteLine("Error during ripping, extraction aborted");
+			Console.WriteLine("See this message below for details:");
+			Console.WriteLine(ex.Message);
 #endif
 		}
 
-		private static void PrepareExportDirectory(string path)
+		public static void ParseArgumentsAndRun(string[] args)
 		{
-			if (Directory.Exists(path))
+			Ripper ripper = new();
+			ripper.Settings.ResetToDefaultValues();
+
+			RootCommand rootCommand = new RootCommand();
```

Comment:
This deviates from our discussions. We talked about a check for when `args` had length 1.

Reply:
This actually does do that, but using System.CommandLine properly, instead of just counting args length, which would be quite hacky since we have System.CommandLine already. If there is no subcommand parsed, the root command is ran, which is exactly that behaviour you described.

https://docs.microsoft.com/en-us/dotnet/standard/commandline/define-commands#define-a-root-command